cmd/observer/node_builder/observer_builder.go

```diff
@@ -615,6 +645,9 @@ func (builder *ObserverServiceBuilder) validateParams() error {
 	if len(builder.bootstrapNodeAddresses) != len(builder.bootstrapNodePublicKeys) {
 		return errors.New("number of bootstrap node addresses and public keys should match")
 	}
+	if len(builder.upstreamNodeAddresses) != len(builder.upstreamIdentities) {
+		return errors.New("number of upstream node addresses and public keys should match")
+	}
```

Comment:
did you mean `upstreamNodePublicKeys` here?

based on my other comments, maybe this should be
```suggestion
	if len(builder.upstreamNodePublicKeys) > 0 && len(builder.upstreamNodeAddresses) != len(builder.upstreamNodePublicKeys) {
		return errors.New("number of upstream node addresses and public keys must match if public keys given")
	}
```

Reply:
Fixed: https://github.com/GetElastech/flow-go/commit/e867cb36f4fde512a7b0cd4d15f6755a46473939